### ISSUE TITLE
Accessing `blendphase` plugin parameter raises `IndexError`

### ISSUE THREAD

**Issue by leroyvn** [mitsuba-renderer/mitsuba3#344]
## Summary

Access to the `weight.data` parameter of a `blendphase` plugin instance raises an `IndexError`.

## System configuration

```
System information:

  OS: Ubuntu 22.04.1 LTS
  CPU: AMD Ryzen 9 5900X 12-Core Processor
  GPU: NVIDIA GeForce GT 730
  Python: 3.10.5 | packaged by conda-forge | (main, Jun 14 2022, 07:04:59) [GCC 10.3.0]
  NVidia driver: 470.141.03
  LLVM: 14.0.0

  Dr.Jit: 0.2.2
  Mitsuba: 3.0.2
     Is custom build? True
     Compiled with: Clang 11.1.0
     Variants:
        scalar_rgb
        scalar_spectral
        llvm_rgb
        llvm_ad_rgb
```

## Description

An `IndexError` is raised upon accessing the `weight.data` parameter of a `blendphase` plugin. The weight table is defined using a `gridvolume`. Although I can't point the exact commit where this regression occurred, I think it was between d456826ef9a3a2309dabd6d821b94e8daa620ba2 and now (7ca09a3ad95cec306c538493fa8450a096560891 behaves as expected).

## Steps to reproduce

1. Compile latest `master` with default variants
2. Run the following script:

`blendphase_params.py`
```python
import numpy as np

import mitsuba as mi

mi.set_variant("scalar_rgb")

phase_mixed = mi.load_dict({
    "type": "blendphase",
    "weight": {"type": "gridvolume", "grid": mi.VolumeGrid(np.full((3, 3, 3), 0.5))},
    "phase_0": {"type": "isotropic"},
    "phase_1": {"type": "rayleigh"},
})

params = mi.traverse(phase_mixed)
weights = params["weight.data"]  # Crashes here
```

Traceback
```
Traceback (most recent call last):
  File "/home/leroyv/Documents/src/thirdparty/mitsuba3/playgrounds/blendphase_params.py", line 15, in <module>
    weights = params["weight.data"]  # Crashes here
  File "/home/leroyv/Documents/src/thirdparty/mitsuba3/build/python/mitsuba/python/util.py", line 55, in __getitem__
    self.update_candidates[key] = _jit_id_hash(value)
  File "/home/leroyv/Documents/src/thirdparty/mitsuba3/build/python/mitsuba/python/util.py", line 286, in _jit_id_hash
    return hash(tuple(jit_ids(value)))
  File "/home/leroyv/Documents/src/thirdparty/mitsuba3/build/python/mitsuba/python/util.py", line 273, in jit_ids
    ids.extend(jit_ids(value[i]))
  File "/home/leroyv/Documents/src/thirdparty/mitsuba3/build/python/mitsuba/python/util.py", line 273, in jit_ids
    ids.extend(jit_ids(value[i]))
  File "/home/leroyv/Documents/src/thirdparty/mitsuba3/build/python/mitsuba/python/util.py", line 273, in jit_ids
    ids.extend(jit_ids(value[i]))
  [Previous line repeated 1 more time]
  File "/home/leroyv/Documents/src/thirdparty/mitsuba3/build/python/drjit/detail.py", line 693, in tensor_getitem
    shape, index = slice_tensor(tensor.shape, slice_arg, tensor_t.Index)
  File "/home/leroyv/Documents/src/thirdparty/mitsuba3/build/python/drjit/detail.py", line 603, in slice_tensor
    raise IndexError("slice_tensor(): too many indices specified!")
```

</details>

**Comment by Speierers** [comment]
Thanks for reporting this issue, it should be fixed in now: [a0101e9](https://github.com/mitsuba-renderer/mitsuba3/commit/a0101e97adba10a1d997a87afd70245139dfa41f)

**Comment by leroyvn** [comment]
Thank you!
